content/3_Silver/2P.mdx

```diff
@@ -31,6 +31,118 @@ frequency: 2
 	</Resource>
 </Resources>
 
+## Two Pointers
+
+The two pointers method iterates two pointers across an array, to track the start and end
+of an interval. It can also be used to track two values in an array as shown in [CPH's 2SUM](https://usaco.guide/CPH.pdf#page=87) solution. 
+
+<FocusProblem problem="book" />
+
+## Solution - Books
+
+We want to find the longest contiguous segment of books that can be read within $t$ minutes. 
+
+To accomplish this, we can define $\texttt{left}$ and $\texttt{right}$ as the start and end of the segment respectively and use two pointers to track the interval.
+
+For every $\texttt{left}$ index, we need to increase our $\texttt{right}$ pointer until the total time for that segment exceeds $t$.
+
+We will keep track of the max segment size with $\texttt{ans}$, and take the maximum $\texttt{right} - \texttt{left}$ value.
+
+We can then continually increment $\texttt{left}$ by 1.
+
+**The $\texttt{right}$ pointer does not restart after an iteration, it simply keeps moving right; therefore, since both pointers will move at most $N$ times, the overall time complexity is $\mathcal{O}(N)$.**
```

Comment:
I would have said something like "At each timestep, exactly one pointer is incremented. Since each pointer can be incremented at most N times, it takes at most 2N time, resulting in a time complexity of $\mathcal{O}(N)$.
```suggestion
**The $\texttt{right}$ pointer does not restart after an iteration, it simply keeps moving right; therefore, since both pointers will move at most $N$ times, the overall time complexity is $\mathcal{O}(N)$.**
```

Reply:
but what you said is fine as well